shared/getErrorMessage.ts

```diff
@@ -0,0 +1,32 @@
+/**
+ * @see https://kentcdodds.com/blog/get-a-catch-block-error-message-with-typescript
+ */
+
+type ErrorWithMessage = {
+	message: string;
+};
+
+function isErrorWithMessage(error: unknown): error is ErrorWithMessage {
+	return (
+		typeof error === 'object' &&
+		error !== null &&
+		'message' in error &&
+		typeof (error as Record<string, unknown>).message === 'string'
+	);
+}
+
+function toErrorWithMessage(maybeError: unknown): ErrorWithMessage {
+	if (isErrorWithMessage(maybeError)) return maybeError;
+
+	try {
+		return new Error(JSON.stringify(maybeError));
```

Comment:
maybe a `if (typeof maybeError === 'string') return new Error(maybeError);`? to avoid unnecessary quoting - not that I can see anything that would cause this